### PR TITLE
Remove xenobotany gene masks.

### PR DESCRIPTION
**Pull request by PJB3005**
They added nothing good to the game.

Adapted from https://github.com/ParadiseSS13/Paradise/pull/3831

@Mijhead

### CODE REVIEW

**Comment by 9600bauds** [comment]
...?
So you want everything to be braindead simple roundstart because ????????
I even added surefire ways to figure out the genes by being smart with the plants you're using for research instead of trial and error, come on.

File: html/changelogs/PJB3005-phyto.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+author: PJB3005
+delete-after: true
+changes:
+  - rscadd: Removes gene masks from the xenobotany machinery. No more wasting your fucking time on discovering what the genes do.
```
Comment (9600bauds) [inline html/changelogs/PJB3005-phyto.yml:4]: >wasting your fucking time
wew someone feels strongly about this

**Comment by PJB3005** [comment]
Sitting behind the machines for half an hour is very good gameplay yes.

Xenobotany shouldn't have such a long setup time, that's my issue with it.

File: code/modules/hydroponics/seed_controller.dm [hunk 2]
```diff
@@ -8,26 +8,26 @@
 /client/proc/show_plant_genes()
```
Comment (9600bauds) [inline code/modules/hydroponics/seed_controller.dm]: What is even the point of keeping this verb anymore?

Comment (PJB3005) [inline code/modules/hydroponics/seed_controller.dm]: *shrug

It couldn't even be accessed before either.

**Comment by 9600bauds** [comment]
>half an hour 
>half an hour
>THIRTY MINUTES TO FIGURE OUT THE GENES YOU CAN IMMEDIATELY FIGURE OUT WITH ONE SPLICE EACH

Christ fucking almighty and lords above

**Comment by PJB3005** [comment]
Fine 10 minutes. It feels like a goddamn eternity and it's awful.

**Comment by 9600bauds** [comment]
I'm at a loss for words really I don't even know what to say here

**Comment by PJB3005** [comment]
Sure if we can get a BETTER way to research the blocks that's fine but sitting behind the splicer is literally the shittiest way to do it. Complexity for the sake of complexity is not a good thing.

**Comment by PJB3005** [comment]
I wouldn't have a problem with it if the researching wasn't just a 2x worse version of R&D. But it is.

**Comment by 9600bauds** [comment]
>just a 2x worse version of R&D
I don't know how to approach you here because some reason you're fuming about this and not thinking straight
Start with mutated or radically different plants and you don't need to sit behind the splicer nearly as much
You don't even need to figure out all the genes to do a variety of things, even though it's piss fucking easy

**Comment by PJB3005** [comment]
Yes, it IS piss fucking easy if you know what you're doing. Which means it adds literally fucking nothing but wasting your time for a few minutes.

Repetition between rounds is unavoidable. But when it is  _exactly_ the same each time it adds nothing but boredom.

**Comment by 9600bauds** [comment]
There's different ways to do it and part of learning xenobotany is figuring out those tricks. You discover the genes at a random order every shift so you can start doing different plants earlier on.
If you mean "I already figured out the best way to do this and will do it in one single stretch at the start of every shift and hate every second of it" then yes there is no implementation of research you wouldn't get tired of. 
Ripping it all out altogether without even a single thought of how to make it better is such a shitfit, and these outlandish claims you're throwing around really worry me.

**Comment by 9600bauds** [comment]
I mean if we're going by this logic of repetition is bad let's just rip it all off we would have chemfridges that stock themselves, genetics with all the powers mapped out roundstart, xenobio with all the slime colors spawned in, singulo already running roundstart with chilled plasma...

**Comment by PJB3005** [comment]
To quote @FalseIncarnate:

> Xenobotany has never been researching the genes, but rather researching new ways to combine them via hybridization. The only real unknowns are the traits of the randomly generated strange seeds.
> 
> Obviously the contents of the genes are something you can learn via experimentation and/or the code, and previously you had to spend time figuring out which button the genes were on. There was no research to be had in that, only a horribly watered-down game of Simon.

If there was more interest in xeno botany then there would be the seeds you had to use on the wiki's xeno botany page, I guarantee you.

If you want to add a learning curve to xenobotany this is an awful place to put it. There are much better places to put learning curves. Even the tricks and such on how to most effectively do botany are a learning curve and they are way more fun and interesting.

**Comment by 9600bauds** [comment]
Like?
If you want to improve it I'm listening, ripping it all out is a shitfit.

**Comment by PJB3005** [comment]
- Add more genes.
- Add more fun combinations of genes.
- Interactions between genes and chemicals.

And no I don't think that until such a day comes we should keep gene masks.

**Comment by 9600bauds** [comment]
...That's all just more content, which would also be immediately figured out and stuck up on the wiki page if anyone cared by your own admission.

**Comment by 9600bauds** [comment]
Not that more content is bad, but the bottleneck with content is people willing to code it.

**Comment by PJB3005** [comment]
Okay fine.

**Comment by kol1th** [comment]
Let's remove chem masters and chem dispensers, and have infinite of every pill in the fridge.

**Comment by Kurfursten** [comment]
> Pjb, who plays engineering exclusively, complaining about a repetitive task that "you do at the start of every shift that is braindead once you learn how to do it one time, and takes about 10 minutes"

Something about this story doesn't add up...

**Comment by FalseIncarnate** [comment]
I got pulled here from a mention, so I guess it would be polite to stop in and say hi.

I'm not sure how different your botany stuff is from ours, but we removed the masking because it was simply complexity for the sake of complexity. It also meant fewer players were willing to even touch the xenobotany machines, which meant the cool features it offered were utilized by maybe a handful of players much like reconfiguring atmospherics (the work of wizards to some). The removal of the masking was to make the mechanics more accessible to newer botanists. Obviously knowing what is tied to each gene still requires that initial experimentation (or code browsing), but now it just doesn't require you to spend that time each shift, when you could be using what you learn to bury the chef in more fresh produce than he can deep fry into oblivion and have time left to create the perfect death nettle to use before the shuttle is called.

**Comment by Kurfursten** [comment]
I know this is veering off topic but in an effort to figure out who you were I checked out your PR history

> Tablecrafting sandwiches and salads instead of using the microwave

Ooh that's neat. I would port that hell out of that if grad school wasn't picking back up. Kitchen is overdue for some love on /vg/.

**Comment by FalseIncarnate** [comment]
I've done a number of PRs on Paradise for botany and the kitchen, since they are so closely entwined. A buff to botany like this tends to result in an increase in chef player happiness and willingness to try new things, because an experienced botanist is able to supply them more variety with better efficiency.

In case you don't have it already, I suggest considering porting / incorporating https://github.com/ParadiseSS13/Paradise/pull/4488 if you decide to keep this open / merge it, since you'll be seeing a lot more versions of wheat than ever before. If you've got this already, great job.

**Comment by Kurfursten** [comment]
Bauds reworked naming conventions with botany3, but I don't think we had your spam issue even before that - someone perhaps fixed it a while ago. We do have other prefixes though, based on genes, like

> Slippery [plant]

**Comment by FalseIncarnate** [comment]
Probably fine then. We did have that issue (hence the PR) which got real bad when we had the demasking and TG's new bees, resulting in horrifically long smart fridge lists.

**Comment by 9600bauds** [comment]
I had a neat idea thinking about this
Get rid of the disks and the gene masks entirely, have the transfer of genes require two live plants.
So now you got to keep a physical record of each beneficial gene by tending to the "ideal" plants, or keeping their seeds around and planting them and waiting a little while before you can use their genes.
Complexity becomes complexity by execution, the flexibility of which genes you can use depends on which ones you want to keep around. And how good you can organize them
That'd be cool, I think

**Comment by PJB3005** [comment]
It could work but the issue lies with exotic seeds which are too hard to get the conditions right for to actually plant.

**Comment by 9600bauds** [comment]
Totally feature
In fact write a crude system where the trickiness of exotic seed environment conditions is directly related to their number of desirable features

**Comment by PJB3005** [comment]
Sounds good.

**Comment by clusterfack** [comment]
I don't really care either way I've never played xenobotany heavily, so you guys form a concensus amongst yourselves and do what you guys agree on.

**Comment by 9600bauds** [comment]
Thinking about it some more, the "genetic decay" thingy from the machines would finally matter at all, you could tie it to the live plants and force a large amount of modifications to be done throughout multiple plant generations.

**Comment by ComicIronic** [comment]
Server vote this

**Comment by Killette2** [comment]
I don't have a problem with it that much, but the mystery behind it to most players prevents the chef from being a dick and using the machines becoming mandatory.

The only other thing I have to say is it's not that many genes, and would be a lot more justifiable if we had more xenobotany traits. If none are added, this doesn't do that much.

---

## Edit: Fuck this, push it in. It is a total waste of time on second thought.

**Comment by ScrambledSteak** [comment]
FIX
ALL
CONFLICTS

**Comment by ghost** [comment]
Instead of removing it, what if we had the option to assign those tags in the interface to the gene blocks?  You know, the way genetics was made nicer, and the way cult research works with notes?  Maybe the gene splicing/purging times could be reduced along with the gene extraction times?  

Edit: I take it back after consideration.  This is an excellent change and should be merged immediately.  The current system is completely intolerable.

**Comment by PJB3005** [comment]
That's not the reason I'm removing it, it's because it adds a ton of setup to an already unrewarded job.

**Comment by gbasood** [comment]
COOOOOOOONFLICTS

**Comment by PJB3005** [comment]
See I'd fix conflicts if I could get a consensus on whether or not this would ever be merged.

**Comment by Exxion** [comment]
Server poll

**Comment by ammy55** [comment]
On the one hand i would appreciate this to cut down on the time it takes to dive into plant autism. On the other hand though i spent 3 hours with the code and a spreadsheet figuring out the easiest way to map genes and i'll miss having that exclusive leg up.

**Comment by clusterfack** [comment]
Well if people are making spreadsheets just to be able to accomplish this just remove it. It'll end up on the wiki or posted in the thread once a week and then everyone will just use that for powergamin secrets which are no fun for newcomers.

**Comment by Kurfursten** [comment]
People make spreadsheets for bombs and SME setups

**Comment by clusterfack** [comment]
Oh please those things are dead simple

**Comment by Kurfursten** [comment]
Finding masks is actually easier than learning to run an SME. The only difference is that you have personally learned to engineer and not botanist.

**Comment by clusterfack** [comment]
That's not really the case, an SME requires zero prior knowledge whatsoever to make.

**Comment by PJB3005** [comment]
If you have no idea what you're doing with masks you'll be wasting hours getting nowhere.

But my point stands. If you're making a spreadsheet it's only an annoyance and a chore. If you're new it's a source of infuriation and makes less people play the job. If there's a timegate needed this isn't how it should be done.

**Comment by Kurfursten** [comment]
It's just so surreal because I know both of you play engineer, and @PJB3005 you literally yelled at me once for trying to set up the RUST without having "read the code" because there was no wiki page back then. Once you've actually learned botany, finding masks is no more difficult than a routine engine setup or cooling plasma. You can "waste your time for hours getting nowhere" on anything, even atmosia.

My point stands too, people make spreadsheets for everything. Just because one person anecdotally made a spreadsheet, what does that even mean? SarahJohnson famously had a spreadsheet for tox bombs.

**Comment by PJB3005** [comment]
> and @PJB3005 you literally yelled at me once for trying to set up the RUST without having "read the code" because there was no wiki page back then.

When was that? The wiki page was written last year.

There's also a severe difference between having a spreadsheet to do a job and needing a spreadsheet to BEGIN your job.

**Comment by Kurfursten** [comment]
You don't need a spreadsheet to do xenobotany. I don't know where you're getting that impression other than the offhanded comment of one person.

> When was that

It must have been about six months before that because it was around the time that I started harassing you to make the page, and it took a while to appear

**Comment by Intigracy** [comment]
confrictu

**Comment by PJB3005** [comment]
As I said there's no point fixing conflicts if we can't come to an agreement.

**Comment by Killette2** [comment]
@PJB3005 xenobotany wasn't unrewarding, take it back!

**Comment by ComicIronic** [comment]
RIP this confrictu PR
A new PR removing masks will be merged ASAP.

**Comment by PJB3005** [comment]
As I said I would've instantly fixed conflicts if I could have gotten a consent on whether it's gonna ever be merged or not.

If you told me you'd instantly merge it if I fixed conflicts we'd have this merged ages ago.

**Comment by PJB3005** [comment]
Ready for merge as I test it to make sure everything still works.

**Comment by Kurfursten** [comment]
muh redditvotes tho

**Comment by PJB3005** [comment]
And we're good. Ready to merge when you want @ComicIronic.

**Comment by ComicIronic** [comment]
To clarify for people who will want to know the final decision: repetitive design which doesn't add complexity is not desirable. This is exactly the problem with research as it currently stands, and I don't see any reason to keep it around.

**Comment by FantasticFwoosh** [comment]
In my own opinion if you remove it and port a machine similar to TG's gene editor (which does the same thing with disks, though your system is more complex) and outright replace your xenobotany branch of research with something else you might be able to keep the functions of GM editing crops.

> Its still a repetitive process but you wouldn't need to have a entire job role there when you can instead just leave it up to the hydroponics staff who are going to use the hybrid results and grow them as they wish as the primary supplier anyway.

> TG's gene edit system simply consumes a seed to save one of its traits onto a disk to be interchangable with other plants (50 potency extracted at cost of seed = 50 potency given to a new plant seed = disk can be reused multiple times for spreading 50 potency or overwritten) , your lysis - machine there seems to run off a similar system anyway (**though i dont particularly have any firsthand experience of it**)

Aquaculture/botany for fishing and kelp in big water tanks, something specialized for your server and love of complex monitoring machines?  #11929

(anybody who knows the hell of keeping tropical fish will understand)

**Comment by 9600bauds** [comment]
Yes, let's remove xenobotany and replace it with exactly the same thing.

**Comment by FantasticFwoosh** [comment]
Well do you REALLY want to replace it? Then you fully have to commit to scrapping the notion of gene modification.

What i was suggesting was just to downsize it to a piece of hardware put in or close to hydroponics and leave that there, then re-use the space for something else or another science project.

**Comment by Exxion** [comment]
No
He was joking because what you suggested we change it to sounds exactly like what we have now

**Pull request merged**